core/settings.py

```diff
@@ -216,20 +220,193 @@ def as_dict(self):
         """
         d = {}
         for k, v in self.items():
-            if not isinstance(v, Settings):
-                d[k] = v
-            else:
+            if isinstance(v, Settings):
                 d[k] = v.as_dict()
+            elif isinstance(v, list):
+                d[k] = [i.as_dict() if isinstance(i, Settings) else i for i in v]
+            else:
+                d[k] = v
 
         return d
 
 
+    @classmethod
+    def enable_missing(cls):
```

Comment:
This thing is super cool, but please please change the name ;P If you call it "a context manager for temporary disabling the `__missing__` magic method", then why on earth is it called `enable_missing`? ^^

`disable_autocomplete` maybe? or just `disable_missing`?